src/kernel/vfs/vfs.h

```diff
@@ -7,9 +7,27 @@
 
 // DEBUG for testing CMake configuration
 void vfsmain();
+void vfs_init();
 
-// TODO: Implement
-#define ServiceHandle_t void* // placeholder for now
+
+#define ServiceHandle_t struct FtEntry_t* 
+
+//Usually, service handle is a pointer to a file table entry which corresponds
+//to service's communication file. The server-fs driver is the only server 
+//that doesn't have one (because server-fs driver is the guy who manages
+//calls like read or write to communication files, and if it has one too, then
+//someone else would have to manage calls to that file). So we place a random number here.
+
+#define SERVER_FS_HANDLE (FtEntry_t*) 39218098419283132
+
+//Table of opened files
```

Comment:
nit: it does not seem to  be a table – all processes will have a table of `FtEntry_t` objects

Reply:
Not necessarily. Each process will have a table of file descriptors (which, as I thought, will be translated into FtEntry pointers by procfs). FtEntry is our internal representation of opened file, and I've made it global for now, as we don't have a process abstraction yet. It is up to decision whether it should remain global in the future.